### PR TITLE
BG-79275-agoric-sdk-addition

### PR DESCRIPTION
**Pull request by DinshawKothari**
Ticket: BG-79275

### CODE REVIEW

File: modules/sdk-coin-bld/src/bld.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { CosmosCoin } from '@bitgo/abstract-cosmos';
+import { BaseCoin, BitGoBase, Environments } from '@bitgo/sdk-core';
+import { BaseCoin as StaticsBaseCoin, coins } from '@bitgo/statics';
+
+import { TransactionBuilderFactory } from './lib';
+import utils from './lib/utils';
+
+export class Bld extends CosmosCoin {
```
Comment (OttoAllmendinger) [inline modules/sdk-coin-bld/src/bld.ts:8]: please explain acronyms

File: modules/sdk-coin-bld/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+const sdk = new BitGoAPI();
+
+sdk.register('bld', Bld.createInstance);
+```
+
+## Development
+
+Most of the coin implementations are derived from `@bitgo/sdk-core`, `@bitgo/statics`, and coin specific packages. These implementations are used to interact with the BitGo API and BitGo platform services.
+
+You will notice that the basic version of common class extensions have been provided to you and must be resolved before the package build will succeed. Upon initiation of a given SDK coin, you will need to verify that your coin has been included in the root `tsconfig.packages.json` and that the linting, formatting, and testing succeeds when run both within the coin and from the root of BitGoJS.
```
Comment (OttoAllmendinger) [inline modules/sdk-coin-bld/README.md:30]: what is "bld"? please add links to resources

File: modules/sdk-coin-bld/src/bld.ts (hunk 1 above)
Comment (DinshawKothari) [reply modules/sdk-coin-bld/src/bld.ts:8]: Bld is the Agoric chain coin symbol.

File: modules/sdk-coin-bld/README.md (hunk 2 above)
Comment (DinshawKothari) [reply modules/sdk-coin-bld/README.md:30]: Bld is Agoric chain coin symbol.

File: modules/sdk-coin-bld/src/lib/constants.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export const validDenoms = ['nbld', 'ubld', 'mbld', 'bld'];
+export const accountAddressRegex = /^(agoric)1(['qpzry9x8gf2tvdw0s3jn54khce6mua7l']+)$/;
+export const validatorAddressRegex = /^(agoricvaloper)1(['qpzry9x8gf2tvdw0s3jn54khce6mua7l']+)$/;
```
[the remark this answers is not in the document]
Comment (DinshawKothari) [reply modules/sdk-coin-bld/src/lib/constants.ts:3]: Yes, the prefix is different for different chains. Currently adding it in the coin sdks only.
Later we can think of optimising this more may be.

File: modules/sdk-coin-bld/test/integration/index.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+describe('BLD', function () {
+  let bitgo: TestBitGoAPI;
+  before(function () {
+    bitgo = TestBitGo.decorate(BitGoAPI, { env: 'mock' });
+    bitgo.safeRegister('bld', Bld.createInstance);
+    bitgo.safeRegister('tbld', Tbld.createInstance);
+    bitgo.initializeTestVars();
+  });
+});
```
Comment (Vijay-Jagannathan) [inline modules/sdk-coin-bld/test/integration/index.ts:16]: Do we need this ?

Comment (DinshawKothari) [reply modules/sdk-coin-bld/test/integration/index.ts:16]: We don't have much clarity on integration tests now. So same as all other coins, we are adding this here, but in future, we need some better strategy to have better integration testcases.

**Review by Vijay-Jagannathan — APPROVED** [review]

File: modules/sdk-coin-bld/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    {
+      "path": "../sdk-api"
+    },
+    {
+      "path": "../sdk-core"
+    },
+    {
+      "path": "../sdk-test"
+    },
+    {
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-bld/tsconfig.json:22]: @DinshawKothari Missing statics

**Pull request merged**